### PR TITLE
GEOMESA-1384 Verifying calls to geometry.getCentroid

### PR DESCRIPTION
**Pull request by elahrvivaz**
Note: this bug is likely to creep back in unless we are super-vigilant about not calling getCentroid. Probably it should be fixed in geotools.

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-tools/conf/sfts/twitter/reference.conf [hunk 1]
```diff
@@ -177,7 +177,7 @@ geomesa  {
 
         { name = polyWkt, transform = "concat( 'polygon((', mkstring(',', $c0, $c1, $c2, $c3, $c4), '))')"}
 
-        { name = geom, transform = "cql:centroid(polygon($polyWkt))" }
+        { name = geom, transform = "cql:centroid(cql:envelope(polygon($polyWkt)))" }
```
Comment (jnh5y) [inline geomesa-tools/conf/sfts/twitter/reference.conf]: @jahhulbert-ccri this is the fix I was suggesting to you.

**Comment by jnh5y** [comment]
+1

**Comment by jahhulbert-ccri** [comment]
looks good to me. yeah I think we should really contact geotools and put in a ticket to remove this when its fixed since its not intuitive to call envelope before centroid...

**Comment by jnh5y** [comment]
@jahhulbert-ccri I agree.  The problem is actually in getCentroid in JTS.  I've created https://geomesa.atlassian.net/browse/GEOMESA-1387 and assigned it to myself.  This workaround will be in place for awhile...

**Comment by jahhulbert-ccri** [comment]
cool, i'll merge it...anything else we need that you know of?

**Comment by jnh5y** [comment]
Thanks, any additional testing before the release would be great; I don't think we've changed anything 'dangerous' since Friday...

**Comment by jahhulbert-ccri** [comment]
yup i will rerun...i'm doing backwards compat upgrade testing...so it kinda implicitly tests some stuff

**Pull request merged**